### PR TITLE
Add a token generator in secret agent.

### LINKED ISSUE #8844: prow: Expose a secret getter closure for the secret agent.
related to #8579

### PR DESCRIPTION
**Pull request by droslean**
closes #8844

### CODE REVIEW

**Comment by kargakis** [comment]
/lgtm

needs an approval

File: prow/config/secrets_agent.go [hunk 1]
```diff
@@ -95,3 +95,10 @@ func (sa *SecretAgent) SetSecret(secretPath string, secretValue []byte) {
 	defer sa.Unlock()
 	sa.secretsMap[secretPath] = secretValue
 }
+
+// GetTokenGenerator returns a function that gets the value of a given secret.
+func (sa *SecretAgent) GetTokenGenerator(secretPath string) func() []byte {
```
Comment (stevekuznetsov) [inline prow/config/secrets_agent.go]: I might call this `TokenGenerator()` but that's a nit

**Pull request merged**